src/Castle.Windsor.Extensions.MsDependencyInjection/WindsorScopedServiceProvider.cs

```diff
@@ -0,0 +1,93 @@
+// Copyright 2004-2020 Castle Project - http://www.castleproject.org/
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Castle.Windsor.Extensions.MsDependencyInjection
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Reflection;
+    
+    using Castle.Windsor;
+    
+    using Microsoft.Extensions.DependencyInjection;
+
+    internal class WindsorScopedServiceProvider : IServiceProvider, ISupportRequiredService, IDisposable
+    {
+        private readonly NetCoreScope scope;
+        private bool disposing = false;
+
+        private readonly IWindsorContainer container;
+        
+        public WindsorScopedServiceProvider(IWindsorContainer container)
+        {
+            this.container = container;
+            this.scope = NetCoreScope.Current;
+        }
+
+        public object GetService(Type serviceType)
+        {
+            using(var fs = new NetCoreScope.ForcedScope(scope))
+            {
+                return ResolveInstanceOrNull(serviceType, true);    
+            }
+        }
+
+        public object GetRequiredService(Type serviceType)
+        {
+            using(var fs = new NetCoreScope.ForcedScope(scope))
+            {
+                return ResolveInstanceOrNull(serviceType, false);    
+            }
+        }
+
+        public void Dispose()
+        {
+            if(scope is NetCoreRootScope)
+            {
+                if(!disposing)
+                {
+                    disposing = true;
+                    var disposableScope = scope as IDisposable;
+                    if(disposableScope != null)
+                    {
+                        disposableScope.Dispose();
+                    }
```

Comment:
`(scope as IDisposable)?.Dispose();` afaik these constructs are not allowed in Castle

Reply:
The [Castle Coding Standards](https://github.com/castleproject/Home/blob/master/coding-standards.md) document was written over 10 years ago with very little change. If you believe something guides you that this isn't allowed let's change the document because I think this is perfectly acceptable modern C#.